test/extended/router/weighted.go

```diff
@@ -33,10 +31,10 @@ var _ = g.Describe("[Conformance][Area:Networking][Feature:Router] weighted open
 		if len(imagePrefix) == 0 {
 			imagePrefix = "openshift/origin"
 		}
+		saName := fmt.Sprintf("system:serviceaccount:%s:default", oc.Namespace())
 		err := oc.AsAdmin().Run("adm").Args("policy", "add-cluster-role-to-user", "system:router", oc.Username()).Execute()
 		o.Expect(err).NotTo(o.HaveOccurred())
-		// Wait for the policy to be propagated
-		testutil.WaitForClusterPolicyUpdate(oc.InternalKubeClient().Authorization(), "get", api.Resource("service"), true)
+		err = oc.AsAdmin().Run("adm").Args("policy", "add-cluster-role-to-user", "system:router", saName).Execute()
```

Comment:
@smarterclayton it's your test.  is this permission grant correct?

Reply:
No, this is supposed to be testing a router scoped to a single namespace.  That router isn't supposed to need anything above and beyond access to routes.  Giving it a cluster role would be wrong.  I'm not actually following the flake - the errors in the router log are spurious and are expected.  The bug is that the router shouldn't print that error so much because it's intentional in this case.

Reply:
After looking at the test it seems like it fails to wait for the `haproxy_server_up` metric for `weightedroute`:

```
# HELP haproxy_server_up Current health status of the server (1 = UP, 0 = DOWN).
# TYPE haproxy_server_up gauge
haproxy_server_up{namespace="",pod="",route="",server="fe_no_sni",service=""} 1
haproxy_server_up{namespace="",pod="",route="",server="fe_sni",service=""} 1
haproxy_server_up{namespace="default",pod="docker-registry-1-4mtq9",route="docker-registry",server="172.16.4.6:5000",service="docker-registry"} 1
haproxy_server_up{namespace="default",pod="registry-console-1-ckmts",route="registry-console",server="172.16.6.2:9090",service="registry-console"} 1
haproxy_server_up{namespace="kube-system",pod="prometheus-0",route="alertmanager",server="172.16.2.51:10443",service="alertmanager"} 1
haproxy_server_up{namespace="kube-system",pod="prometheus-0",route="alerts",server="172.16.2.51:9443",service="alerts"} 1
haproxy_server_up{namespace="kube-system",pod="prometheus-0",route="prometheus",server="172.16.2.51:8443",service="prometheus"} 1
```

Reply:
@smarterclayton i will make the wait longer, but I still think that the informer errors even if they are spurious might contribute to this flake?

Reply:
from what I can see the only variable that might affect this metric is `ROUTER_METRICS_HAPROXY_SERVER_THRESHOLD` ?

Reply:
@mfojtik we hit this limits and except sharding we don't know how to adjust this values also haproxy_server metrics gone after hitting this limits:
```
# HELP haproxy_exporter_server_threshold Number of servers tracked and the current threshold value.
# TYPE haproxy_exporter_server_threshold gauge
haproxy_exporter_server_threshold{type="current"} 1086
haproxy_exporter_server_threshold{type="limit"} 500
```